File: 26/01/3507.cpp
#include <climits>
#include <vector>

class Solution {
public:
    bool isIncrease(std::vector<int>& nums){
        for (int i = 1; i<nums.size(); ++i) {
            if (nums[i] < nums[i-1]) {
                return false;
            }
        }
        return true;
    }

    int minimumPairRemoval(std::vector<int>& nums) {
        int ans = 0;
        while (!isIncrease(nums)) {
            int index = 0;
            int minX = INT_MAX;
            for (int i = 0; i< nums.size()-1; i++) {
                int temp = nums[i]+nums[i+1];
                if (minX > temp) {
                    minX = temp;
                    index = i;
                }
            }
            nums.erase(nums.begin()+index+1);
            nums[index] = minX;
            ans++;
        }
        return ans;
    }
};

File: 26/01/go/3507.go
package A

import "math"

func isIncrease(nums []int) bool {
	for i := range nums {
		if i == len(nums)-1 {
			break
		}
		if nums[i] > nums[i+1] {
			return false
		}
	}
	return true
}

func minimumPairRemoval(nums []int) int {
	ans := 0
	index := 0
	temp := 0
	for !isIncrease(nums) {
		minX := math.MaxInt
		for i := range nums {
			if i == len(nums)-1 {
				break
			}
			temp = nums[i] + nums[i+1]
			if minX > temp {
				minX = temp
				index = i
			}
		}
		nums[index] = minX
		nums = append(nums[:index+1], nums[index+2:]...)
		ans++
	}
	return ans
}
